core/src/main/scala/org/apache/spark/sql/execution/CodegenSparkFallback.scala

```diff
@@ -87,6 +88,15 @@ case class CodegenSparkFallback(var child: SparkPlan) extends UnaryExecNode {
             }
           case None => throw t
         }
+    } finally {
+      // SNAP-1422
+      val cacheMetaDataForConnector =
+        java.lang.Boolean.getBoolean("SMART_CONNECTOR_CACHE_METADATA")
+      lazy val session = sqlContext.sparkSession.asInstanceOf[SnappySession]
+      if (!cacheMetaDataForConnector &&
+        ExternalStoreUtils.isSmartConnectorMode(session.sparkContext)) {
+        session.sessionCatalog.invalidateAll()
```

Comment:
Read through SNAP-1422. This is what Sumedh is saying "There handling for specific exception types can be added which can re-fetch the meta-data in connector mode and then re-execute the entire plan". 

We can not invalidate the catalog after every query execution. We should do it only when we get an exception that too a specific one. And we, retry the query after refreshing the catalog when the exception is received. See the catch part of the CodeGenSparkFallback.